ext/drreg/drreg.c

```diff
@@ -1173,11 +1173,16 @@ drreg_statelessly_restore_app_value(void *drcontext, instrlist_t *ilist, reg_id_
          */
 #ifdef X86
     if (reg != DR_REG_NULL && pt->aflags.xchg == reg) {
+        /* drreg_restore_app_value invokes drreg_move_aflags_from_reg with stateful=false
+         * above. This spills aflags to some slot, regardless of whether they are live or
+         * not. Also, even though aflags are spilled to a slot, pt->aflags.xchg is not
+         * cleared. Below, we restore aflags to xax, and forget the spill slot.
+         */
         ASSERT(pt->aflags.slot != MAX_SPILLS, "Aflags slot not reserved");
         pt->slot_use[pt->aflags.slot] = DR_REG_XAX; /* appease assert */
```

Comment:
Why is this needed?  spil_reg set slot_use for us -- at least at the where_restore point.

Reply:
was trying not to touch this code, thinking that there may be some corner case that needs it. But I don't see why this assert was needed in the first place. I removed it in a clean client, and it didn't hit any assert.

Reply:
Oh, I think some assert may be hit if aflags were spilled using some reg other than xax (which may happen if xax was already spilled before we reserved aflags). I can write a test for this, and maybe workaround so that we always spill from xax (otherwise the restore logic also becomes more complicated).

Reply:
This is the case I was referring to in previous comment: if aflags were spilled using some other reg: https://github.com/DynamoRIO/dynamorio/blob/27e2f830e209d3c2aca725628b6b453ed5d6ad68/ext/drreg/drreg.c#L1441

Reply:
Maybe spill_reg wasn't called at the time this assert was added.

Reply:
For the xax_swap case: you mean the lazy unrestore could have that slot still holding the other reg.  But...it was spilled to a non-aflags slot, so should that xax_swap code go clear the aflags slot?

Reply:
> Oh, I think some assert may be hit if aflags were spilled using some reg other than xax

Actually, this cannot happen. The `drreg_statelessly_restore_app_value` aflags spill via `drreg_restore_app_value(xax)` is always done using xax:
https://github.com/DynamoRIO/dynamorio/blob/27e2f830e209d3c2aca725628b6b453ed5d6ad68/ext/drreg/drreg.c#L1355

Reply:
> For the xax_swap case: you mean the lazy unrestore could have that slot still holding the other reg. But...it was spilled to a non-aflags slot, so should that xax_swap code go clear the aflags slot?

Not sure I understand this scenario. 

Just to be clear, the `xax_swap` case (which happens when aflags are being spilled at a time when xax is already reserved and in-use) and the mysterious "appease assert" statement are not related (they seemed to be to me initially, but as I said in my last comment, they aren't).

Still can't figure out why the "appease assert" stmt was needed; based on what I read in code, it doesn't seem like it could be an issue. Removed it.